standalone-metastore/metastore-server/src/main/java/org/apache/hadoop/hive/metastore/ObjectStore.java

```diff
@@ -10420,6 +10522,100 @@ public boolean deleteTableColumnStatistics(String catName, String dbName, String
     return ret;
   }
 
+  @Override
+  public boolean deleteTableMultiColumnStatistics(String catName, String dbName, String tableName,
+                                             List<String> colNames, String engine)
+          throws NoSuchObjectException, MetaException, InvalidObjectException, InvalidInputException {
+    boolean ret = false;
+    Query query = null;
+    dbName = org.apache.commons.lang3.StringUtils.defaultString(dbName,
+            Warehouse.DEFAULT_DATABASE_NAME);
+    if (tableName == null) {
+      throw new InvalidInputException("Table name is null.");
+    }
+    try {
+      openTransaction();
+      MTable mTable = getMTable(catName, dbName, tableName);
+      MTableColumnStatistics mStatsObj;
+      List<MTableColumnStatistics> mStatsObjColl;
+      if (mTable == null) {
+        throw new NoSuchObjectException("Table " +
+                TableName.getQualified(catName, dbName, tableName)
+                + "  for which stats deletion is requested doesn't exist");
+      }
+      // Note: this does not verify ACID state; called internally when removing cols/etc.
+      //       Also called via an unused metastore API that checks for ACID tables.
+      query = pm.newQuery(MTableColumnStatistics.class);
+      String filter;
+      String parameters;
+      if (colNames != null) {
```

Comment:
nit: `colNames != null and !colNames.isEmpty()`

Reply:
Acknowledged.